lib/utils/include/utils/graph/cow_ptr_t.h

```diff
@@ -90,21 +95,33 @@ struct cow_ptr_t {
     swap(lhs.ptr, rhs.ptr);
   }
 private:
-  void set_shared(shared_t ptr) {
-    this->ptr = variant<shared_t>(std::move(ptr));
+  void set_shared(shared_t ptr) const {
```

Comment:
Why this change?

Reply:
because type of this->ptr is `mutable variant<std::unique_ptr<T>,std::shared_ptr<T const>> `